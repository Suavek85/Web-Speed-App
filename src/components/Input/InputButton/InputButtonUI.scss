@import "../../../scss/styles.scss";

.inputbutton__wrapper {
    display: block;
    padding: 16px;
    margin-bottom: 20px;
    color: white;
    outline: none;
    border: none;
    cursor: pointer;
    &--inactive {
      background-color: rgba(55,71,133,.8);
      &:active {
        transform: translateY(2px);
        background-color:rgba(55,71,133,1)
      }
    }
    &--loading {
        background-color: $color-warning-loading;
    }
    &--error {
      background-color: $color-bottom-background;
    }
    &--success {
      background-color: $color-green-success;
    }
}

:export {
  block: inputbutton__wrapper;
  blockInactive: inputbutton__wrapper--inactive;
  blockLoading: inputbutton__wrapper--loading;
  blockError: inputbutton__wrapper--error;
  blockSuccess: inputbutton__wrapper--success;
}