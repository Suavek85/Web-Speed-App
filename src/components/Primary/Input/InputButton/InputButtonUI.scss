@import '../../../../scss/styles';

$block: inputbutton;
$blockInactive: modifier($block, inactive);
$blockLoading: modifier($block, loading);
$blockError: modifier($block, error);
$blockSuccess: modifier($block, success);
$blockMobile: modifier($block, mobile);

.#{$block} {
  display: block;
  width: 120px;
  height: 44px;
  margin-left: 50px;
  font-size: em(12);
  font-weight: 500;
  line-height: 44px;
  color: #fff;
  text-align: center;
  cursor: pointer;
  border: 0;
  outline: none;

  &--inactive {
    background-color: $color-blue;

    &:active {
      transform: translateY(2px);
    }
  }

  &--loading {
    background-color: $color-warning-loading;
  }

  &--error {
    width: 160px;
    background-color: $color-alert;
  }

  &--success {
    background-color: $color-green-success;
  }

  &--mobile {
    margin-top: 30px;
    margin-left: 0;
  }
}

:export {
  block: $block;
  blockInactive: $blockInactive;
  blockLoading: $blockLoading;
  blockError: $blockError;
  blockSuccess: $blockSuccess;
  blockMobile: $blockMobile;
}
