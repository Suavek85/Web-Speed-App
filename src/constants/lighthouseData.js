export const lighthouseData = [
  'first-contentful-paint',
  'largest-contentful-paint',
  'speed-index',
  'interactive', 
  'total-blocking-time',
  'cumulative-layout-shift'
]

//largest-contentful-paint

//Largest Contentful Paint element
//Largest Contentful Paint marks the time at which the largest text or image is painted.


//first-contentful-paint

//First Contentful Paint marks the time at which the first text or image is painted

//total-blocking-time

//"Sum of all time periods between FCP and Time to Interactive, when task length exceeded 50ms, expressed in milliseconds


//cumulative-layout-shift
//Cumulative Layout Shift measures the movement of visible elements within the viewport.


        