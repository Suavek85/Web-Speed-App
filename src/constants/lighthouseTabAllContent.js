export const TITLE = 'Title:'
export const DESCRIPTION = 'Description:'
export const SCORE = 'Score:'
export const DISPLAY_VALUE = 'Display Value:'