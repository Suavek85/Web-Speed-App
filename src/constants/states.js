const states = [
  'LOADING',
  'INACTIVE',
  'ERROR',
  'SUCCESS'
]
    
export const appStates = states.reduce((acc, next) => {
  acc[next] = next
  return acc
}, {})