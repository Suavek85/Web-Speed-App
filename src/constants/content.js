import { appStates } from './states'

//APP HEADER
export const APPTITLE = 'Web Speed App'
export const SETTINGS = 'Settings'
export const SIGN_IN = 'Sign in'


//BUTTONS
export const btnsContent = {...appStates}
btnsContent.LOADING = 'Loading...' 
btnsContent.INACTIVE = 'Get report' 
btnsContent.ERROR = 'Oops, an error' 
btnsContent.SUCCESS = 'Get new report'

//LIGHTHOUSE

export const LIGHTHOUSE_TITLE = 'Lighthouse Score'
export const PERFORMANCE = 'Performance'

//FIRST CONTENTFUL PAIN
export const FIRST_PAINT_HEADER = 'First Contentful Paint'
export const FIRST_PAINT_CONTENT = 'First Contentful Paint (FCP) measures the time from navigation to the time when the browser renders the first bit of content from the DOM. This is an important milestone for users because it provides feedback that the page is actually loading.'
export const RESULT = 'Result: '
export const INPUT_HINT = 'Enter your website url in this format https://www.bbc.co.uk/'

//FIRST INPUT DELAY
export const FIRST_INPUT_DELAY_HEADER = 'First Input Delay'
export const FIRST_INPUT_DELAY_CONTENT = 'First Input Delay (FID) measures the time from when a user first interacts with your site (i.e. when they click a link, tap on a button, or use a custom, JavaScript-powered control) to the time when the browser is actually able to respond to that interaction.'


//LIGHTHOUSE TABLE

//FIRST PAINT - LIGHTHOUSE
const FIRST_PAINT_TITLE = 'First Contentful Paint'
const FIRST_PAINT_DESC = 'First Contentful Paint marks the time at which the first text or image is painted.'

//LARGEST PAINT - LIGHTHOUSE
const LARGEST_PAINT_TITLE = 'Largest Contentful Paint'
const LARGEST_PAINT_DESC = 'Largest Contentful Paint marks the time at which the largest text or image is painted.'

//SPEED INDEX - LIGHTHOUSE
const SPEED_INDEX_TITLE = 'Speed Index.'
const SPEED_INDEX_DESC = 'Speed Index shows how quickly the contents of a page are visibly populated.'

//TIME TO INTERACTIVE - LIGHTHOUSE
const INTERACTIVE_TITLE = 'Time to Interactive.'
const INTERACTIVE = 'Time to interactive is the amount of time it takes for the page to become fully interactive.'

//TOTAL BLOCKING TIME - LIGHTHOUSE
const BLOCKING_TIME_TITLE = 'Total Blocking Time'
const BLOCKING_TIME_DESC = 'Sum of all time periods between FCP and Time to Interactive, when task length exceeded 50ms, expressed in milliseconds'

//CUMULATIVE LAYOUT SHIFT
const LAYOUT_SHIFT_TITLE = 'Cumulative Layout Shift.'
const LAYOUT_SHIFT_DESC = 'Cumulative Layout Shift measures the movement of visible elements within the viewport.'

//TABS HEADERS
export const tabsHeader = [
  { content: 'First Paint' },
  { content: 'Largest Paint' },
  { content: 'Speed Index' },
  { content: 'Interactive' },
  { content: 'Blocking Time' },
  { content: 'Layout Shift' },
]

//TABS TITLES
export const tabsTitles = [
  FIRST_PAINT_TITLE, 
  LARGEST_PAINT_TITLE, 
  SPEED_INDEX_TITLE, 
  INTERACTIVE_TITLE,
  BLOCKING_TIME_TITLE,
  LAYOUT_SHIFT_TITLE
]

//TABS DESCRIPTIONS
export const tabsDescriptions = [
  FIRST_PAINT_DESC, 
  LARGEST_PAINT_DESC, 
  SPEED_INDEX_DESC, 
  INTERACTIVE,
  BLOCKING_TIME_DESC,
  LAYOUT_SHIFT_DESC
]
