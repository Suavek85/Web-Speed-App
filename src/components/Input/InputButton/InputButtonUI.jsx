import React from 'react'
import PropTypes from 'prop-types'
import cls from 'classnames'

//INTERNAL
import { appStates } from '../../../constants/states'
import { btnsContent } from '../../../constants/content'
import './InputButtonUI.scss'  

function InputButtonUI(props) {
 
  const { getAppState, handleGetReportClick } = props

  //BUTTON FORMATTING - CLASSES AND CONTENT
  const isStatus = state => getAppState === state
  
  const getContentFromStatus = state => { 
    const contentIndex = Object.keys(btnsContent).findIndex(el => el === state)
    return Object.values(btnsContent)[contentIndex]
  }

  const getButtonClasses = cls({
    'inputbutton__wrapper': true,
    'inputbutton__wrapper--inactive': isStatus(appStates.INACTIVE),
    'inputbutton__wrapper--loading': isStatus(appStates.LOADING),
    'inputbutton__wrapper--error': isStatus(appStates.ERROR),
    'inputbutton__wrapper--success': isStatus(appStates.SUCCESS),
  })

  return (
    <>
      <button 
        onClick={handleGetReportClick} 
        disabled={isStatus(appStates.LOADING)}
        className={getButtonClasses}
      >   
        {getContentFromStatus(getAppState)}
      </button>
    </>
  )
}

InputButtonUI.propTypes = {
  data: PropTypes.string,
  getAppState: PropTypes.string,
  handleGetReportClick: PropTypes.func
}

export default InputButtonUI