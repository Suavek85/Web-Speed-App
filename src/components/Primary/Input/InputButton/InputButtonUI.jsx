import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames/bind'

import { appStates } from '../../../../constants/states'
import { btnsContent } from '../../../../constants/content'
import styles from './InputButtonUI.scss'  

export default function InputButtonUI(props) {
 
  const { getAppState, handleGetReportClick } = props

  //BUTTON FORMATTING - CLASSES AND CONTENT
  const isStatus = state => getAppState === state
  
  const getContentFromStatus = state => { 
    const contentIndex = Object.keys(btnsContent).findIndex(el => el === state)
    return Object.values(btnsContent)[contentIndex]
  }

  let cx = classNames.bind(styles)

  let getButtonClasses = cx({
    block: true,
    blockInactive: isStatus(appStates.INACTIVE),
    blockLoading: isStatus(appStates.LOADING),
    blockError: isStatus(appStates.ERROR),
    blockSuccess: isStatus(appStates.SUCCESS),
  })

  return (
    <>
      <button 
        onClick={handleGetReportClick} 
        disabled={isStatus(appStates.LOADING)}
        className={getButtonClasses}
      >   
        {getContentFromStatus(getAppState)}
      </button>
    </>
  )
}

InputButtonUI.propTypes = {
  data: PropTypes.string,
  getAppState: PropTypes.string,
  handleGetReportClick: PropTypes.func
}
