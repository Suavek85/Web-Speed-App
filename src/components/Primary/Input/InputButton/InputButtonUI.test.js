import React from 'react'
import { shallow } from 'enzyme'
import { appStates } from '../../../../constants/states'
import InputButtonUI from './InputButtonUI'

describe('InputButtonUI snapshot', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<InputButtonUI />)
    expect(wrapper).toMatchSnapshot()
  })

})

describe('InputButtonUI rendering', () => {

  it('should add classes for different states', () => {
    const props = { getAppState: appStates.LOADING}
    const wrapper = shallow(<InputButtonUI {...props} />)
    expect(wrapper.find('button.blockLoading').exists()).toBeTruthy()
  })

})