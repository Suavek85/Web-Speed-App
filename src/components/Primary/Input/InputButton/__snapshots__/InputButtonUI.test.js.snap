// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<Fragment>
  <button
    className="block"
    disabled={false}
  />
</Fragment>
`;
